script.rb

```diff
@@ -0,0 +1,81 @@
+module Enumerable
+  def my_each(array)
+    index = 0
+    while index < array.length
+      yield(array[index])
+      index += 1
+    end
+    array
+  end
+
+  def my_each_with_index(array)
+    index = 0
+    while index < array.length
+      yield(array[index], index)
+      index += 1
+    end
+    array
+  end
+
+  def my_select(array)
+    new_arr = []
+    my_each(array) do |x|
+      new_arr.push(x) if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all?(array)
+    p my_select(array) { |x| yield(x) }.length == array.length
+  end
+
+  def my_any?(array)
+    p my_select(array) { |x| yield(x) }.length.positive? ? true : false
+  end
+
+  def my_none?(array)
+    p my_select(array) { |x| yield(x) }.length.length.zero? ? true : false
+  end
+
+  def my_count(array, idx = 0)
+    index = idx
+
+    if block_given? == true
+      my_select(array) { |x| yield(x) }.length
+
+    else
+      array.length - index
+    end
+  end
+
+  def my_map_block(array)
+    new_arr = []
+    my_each(array) do |x|
+      new_arr.push(yield(x))
+    end
+    p new_arr
+  end
+
+  def my_map_proc(array, &block)
+    new_arr = []
+    my_each(array) do |x|
+      new_arr.push(block.call(x))
+    end
+    p new_arr
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

> you should only have one method for my_map instead of 2 methods

#### my_map 
> Remember to test it not only for arrays but for ranges as well

- [ ] given a block returns a new array with the results of running the block once for every element in enum.
```ruby
[2,4,7,11].my_map {|num| num < 10 } == [2,4,7,11].map {|num| num < 10 
```
- Real map
![image](https://user-images.githubusercontent.com/52209067/77697467-85ce4a00-6f74-11ea-9650-70790cae2000.png)

- [ ] returns an Enumerator if no block is given
```ruby
[2,4,7,11].my_map should be equal to [2,4,7,11].map #<Enumerator: [2, 4, 7, 11]:map
```
- Real map
![image](https://user-images.githubusercontent.com/52209067/77697646-dc3b8880-6f74-11ea-9277-4ed4a2b3af4c.png)

- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so let's work on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```